scripts/install-pi-transactionally.sh

```
@@ -23,84 +90,111 @@ home=sys.argv[1]; paths=sys.argv[2:]; out={}
+state=json.load(open(sys.argv[2]))
+for raw in reversed(list(state)):
+ if not state[raw] and os.path.isdir(raw):
+  try: os.rmdir(raw)
+  except OSError: pass
 PY
+}
+
+restore() {
+  local rel after
+  for rel in "${PATHS[@]}"; do
+    rm -rf "$HOME/$rel"
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to / .

(SC2115)